enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -117,27 +124,36 @@ const InteractiveDashboardInner = ({
           questionPath={adhocQuestionUrl}
           withTitle={withTitle}
           height={questionHeight}
-          plugins={questionPlugins}
+          plugins={plugins}
           onNavigateBack={handleNavigateBackToDashboard}
         />
       ) : (
-        <PublicOrEmbeddedDashboard
-          dashboardId={dashboardId}
-          parameterQueryParams={initialParameterValues}
-          hideDownloadButton={displayOptions.hideDownloadButton}
-          hideParameters={displayOptions.hideParameters}
-          titled={displayOptions.titled}
-          cardTitled={withCardTitle}
-          theme={theme}
-          isFullscreen={isFullscreen}
-          onFullscreenChange={onFullscreenChange}
-          refreshPeriod={refreshPeriod}
-          onRefreshPeriodChange={onRefreshPeriodChange}
-          setRefreshElapsedHook={setRefreshElapsedHook}
-          font={font}
-          bordered={displayOptions.bordered}
-          navigateToNewCardFromDashboard={handleNavigateToNewCardFromDashboard}
-        />
+        <InteractiveDashboardProvider
+          plugins={{ ...globalPlugins, ...plugins }}
+          onEditQuestion={(question: Question) =>
+            setAdhocQuestionUrl(Urls.question(question.card()))
+          }
```

Comment:
I leave a comment about this in `enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/context.tsx`. tl;dr we should memoize these values here as well.

Reply:
I don't think we need to memoize the values here - the function takes in the question from the call, and sets the state of the `adhocQuestionUrl` but there's no explicit dependencies here